### PR TITLE
Unity, CSharp improvements

### PR DESCRIPTION
**Pull request by nitreo**
Unity SDK: fixed leak with Unity Web Request
CSharp SDK: reuse HttpClient, Parallel Requests test

### CODE REVIEW

File: targets/csharp/source/source/PlayFabHttp/PlayFabSysHttp.cs [hunk 1]
```diff
@@ -12,6 +11,14 @@ namespace PlayFab.Internal
 {
     public class PlayFabSysHttp : IPlayFabHttp
     {
+		
+		private readonly HttpClient _client;
```
Comment (pgilmorepf) [inline targets/csharp/source/source/PlayFabHttp/PlayFabSysHttp.cs]: Tabbing issues.

File: targets/csharp/source/source/PlayFabHttp/PlayFabSysHttp.cs [hunk 2]
```diff
@@ -12,6 +11,14 @@ namespace PlayFab.Internal
...
+
+        public PlayFabSysHttp()
+        {
+            _client = new HttpClient();
```
Comment (pgilmorepf) [inline targets/csharp/source/source/PlayFabHttp/PlayFabSysHttp.cs]: Constructor not necessary. You can set this on line 15

File: targets/csharp/source/source/PlayFabHttp/PlayFabWinHttp.cs [hunk 3]
```diff
@@ -10,6 +10,14 @@ namespace PlayFab.Internal
 {
     public class PlayFabWinHttp : IPlayFabHttp
     {
+		
+		private readonly HttpClient _client;
```
Comment (pgilmorepf) [inline targets/csharp/source/source/PlayFabHttp/PlayFabWinHttp.cs]: tabbing issues.

File: targets/csharp/source/source/PlayFabHttp/PlayFabWinHttp.cs [hunk 4]
```diff
@@ -10,6 +10,14 @@ namespace PlayFab.Internal
...
+
+        public PlayFabSysHttp()
+        {
+            _client = new HttpClient();
```
Comment (pgilmorepf) [inline targets/csharp/source/source/PlayFabHttp/PlayFabWinHttp.cs]: Constructor not necessary. You can set this on line 14

File: targets/csharp/source/source/Uunit/PlayFabApiTest.cs [hunk 5]
```diff
@@ -262,6 +264,31 @@ private void UserDataApiContinued3(PlayFabResult<GetUserDataResult> getDataResul
             testContext.True(testMin <= timeUpdated && timeUpdated <= testMax);
         }
 
+		
+		
+		/// <summary>
```
Comment (pgilmorepf) [inline targets/csharp/source/source/Uunit/PlayFabApiTest.cs]: tabbing issues.

File: targets/csharp/source/source/Uunit/PlayFabApiTest.cs [hunk 6]
```diff
@@ -262,6 +264,31 @@ private void UserDataApiContinued3(PlayFabResult<GetUserDataResult> getDataResul
...
+        /// </summary>
+        [UUnitTest]
+        public void ParallelRequests(UUnitTestContext testContext)
+        {
+            var tasks = Enumerable.Range(0, 10)
+                .Select(_ => PlayFabClientAPI.GetUserDataAsync(new GetUserDataRequest(), _, extraHeaders).ThrowIfApiError());
+
+            Task.WhenAll(tasks).ContinueWith(token =>
+            {
+                if (!token.IsCanceled && !token.IsFaulted)
```
Comment (pgilmorepf) [inline targets/csharp/source/source/Uunit/PlayFabApiTest.cs]: This is a good debug test, but not a good long term test.
If 9 threads fail, and the last one passes, the state that's recorded will be "passed".
If 5 threads pass, and then 1 fails, and then 4 pass, the state that's recorded will be "passed".
In reality, you should only call "EndTest" once, with the correct state. You're calling "EndTest" ten times.
You'll need to figure out a way to perform this test such that the 10 threads run in parallel, you save the state of each somewhere, and then evaluate all states at the far end, and call EndTest just once.

File: targets/csharp/source/source/Uunit/PlayFabApiTest.cs [hunk 7]
```diff
@@ -419,4 +446,19 @@ public void WriteEvent(UUnitTestContext testContext)
         }
     }
 }
+
+public static class ApiCallExtensions
```
Comment (pgilmorepf) [inline targets/csharp/source/source/Uunit/PlayFabApiTest.cs]: This is unused?

File: targets/unity-v2/source/Shared/Internal/PlayFabHttp/PlayFabUnityHttp.cs [hunk 8]
```diff
@@ -117,6 +117,7 @@ private IEnumerator Post(CallRequestContainer reqContainer)
                 {
                     OnError("Unhandled error in PlayFabWWW: " + e, reqContainer);
                 }
+				www.Dispose();
```
Comment (pgilmorepf) [inline targets/unity-v2/source/Shared/Internal/PlayFabHttp/PlayFabUnityHttp.cs]: tabbing issues.

File: targets/csharp/source/source/PlayFabHttp/PlayFabSysHttp.cs (hunk 2 above)
Comment (nitreo) [reply targets/csharp/source/source/PlayFabHttp/PlayFabSysHttp.cs]: I'll fix that. The point was that the next person to modify HttpClient settings won't have to do this small refactoring.

File: targets/csharp/source/source/Uunit/PlayFabApiTest.cs (hunk 6 above)
Comment (nitreo) [reply targets/csharp/source/source/Uunit/PlayFabApiTest.cs]: I'm not sure what you mean. Continuation runs on "Task.WhenAll" which is only triggered once all tasks complete. WhenAll task is faulted if one of children if faulted.

File: targets/csharp/source/source/Uunit/PlayFabApiTest.cs (hunk 7 above)
Comment (nitreo) [reply targets/csharp/source/source/Uunit/PlayFabApiTest.cs]: It is used. line PlayFabApiTest : 276:
```
var tasks = Enumerable.Range(0, 10)
                .Select(_ => PlayFabClientAPI.GetUserDataAsync(new GetUserDataRequest(), _, extraHeaders).ThrowIfApiError());
```

File: targets/csharp/source/source/PlayFabHttp/PlayFabSysHttp.cs (hunk 1 above)
Comment (nitreo) [reply targets/csharp/source/source/PlayFabHttp/PlayFabSysHttp.cs]: I found the source of those nasty tabbing issues. Will fix em all

Comment (nitreo) [reply targets/csharp/source/source/PlayFabHttp/PlayFabSysHttp.cs]: fixed

File: targets/csharp/source/source/PlayFabHttp/PlayFabSysHttp.cs (hunk 2 above)
Comment (nitreo) [reply targets/csharp/source/source/PlayFabHttp/PlayFabSysHttp.cs]: fixed

File: targets/csharp/source/source/PlayFabHttp/PlayFabWinHttp.cs (hunk 4 above)
Comment (nitreo) [reply targets/csharp/source/source/PlayFabHttp/PlayFabWinHttp.cs]: fixed

File: targets/csharp/source/source/PlayFabHttp/PlayFabWinHttp.cs (hunk 3 above)
Comment (nitreo) [reply targets/csharp/source/source/PlayFabHttp/PlayFabWinHttp.cs]: fixed

File: targets/csharp/source/source/Uunit/PlayFabApiTest.cs (hunk 5 above)
Comment (nitreo) [reply targets/csharp/source/source/Uunit/PlayFabApiTest.cs]: fixed

File: targets/csharp/source/source/Uunit/PlayFabApiTest.cs (hunk 7 above)
Comment (nitreo) [reply targets/csharp/source/source/Uunit/PlayFabApiTest.cs]: made internal

File: targets/unity-v2/source/Shared/Internal/PlayFabHttp/PlayFabUnityHttp.cs (hunk 8 above)
Comment (nitreo) [reply targets/unity-v2/source/Shared/Internal/PlayFabHttp/PlayFabUnityHttp.cs]: fixed

File: targets/csharp/source/source/Uunit/PlayFabApiTest.cs (hunk 6 above)
Comment (pgilmorepf) [reply targets/csharp/source/source/Uunit/PlayFabApiTest.cs]: IC, I misunderstood.

File: targets/unity-v2/source/Shared/Internal/PlayFabHttp/PlayFabUnityHttp.cs [hunk 9]
```diff
@@ -117,6 +117,7 @@ private IEnumerator Post(CallRequestContainer reqContainer)
                 {
                     OnError("Unhandled error in PlayFabWWW: " + e, reqContainer);
                 }
+                www.Dispose();
```
Comment (pgilmorepf) [inline targets/unity-v2/source/Shared/Internal/PlayFabHttp/PlayFabUnityHttp.cs]: Don't commit a CSharpSDK and a UnitySDK change in the same commit.

**Pull request merged**